### PR TITLE
PPC 2.0: Add tests and update existing tests for new checklist

### LINKED ISSUE #8085: PPCv2.0: Add Tests
With the new version of the checklist complete, we should add a new e2e test for it as well as an axe test to karma.

The way this epic came together was rather chaotic, doing these test additions elsewhere didn't make a lot of sense.

### PR DESCRIPTION
**Pull request by samwhale**
## Context

PPC 2.0

## Summary

Add test, fix tests for the new checklist.

## To-do

- [x] fix broken tests
- [x] add axe tests (will fix them here: #8211)
- [x] ~add unit tests for checkpoint provider~ decided against this since all the checkpoints are tested in the integration tests now.
- [x] add more karma tests
- [x] add e2e tests (added [here](https://github.com/google/web-stories-wp/pull/8174/commits/43f7bded58dd06b4b9a031dc9029ef080862804c))

## User-facing changes

none

## Testing Instructions

Run tests! Shuold pass in CI too.

### Does this PR have a security-related impact?

No

### Does this PR change what data or activity we track or use?

No

### Does this PR have a legal-related impact?

No

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This PR contains automated tests (unit, integration, and/or e2e) to verify the code works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #8085

### CODE REVIEW

File: assets/src/edit-story/karma/fixture/containers/checklist.js
Comment (samwhale) [inline assets/src/edit-story/karma/fixture/containers/checklist.js:66]: I needed a way to check to see which tab was actually selected.

I don't love that I added these to the fixture but I figured this was equally as ugly as writing this in the test directly. It's repeated code though, and I'm happy to remove it.

I figured that it made more sense here because the `name` was the same on each. Thoughts?

Comment (BrittanyIRL) [reply assets/src/edit-story/karma/fixture/containers/checklist.js:66]: i like this here because otherwise we repeat it - i was repeating this alot fixing the existing karma tests that i pulled in from the previous structure in #8174

File: assets/src/edit-story/components/checklist/karma/checklist.karma.js
Comment (samwhale) [inline assets/src/edit-story/components/checklist/karma/checklist.karma.js]: fixing these here: #8211

File: assets/src/edit-story/components/checklist/karma/checklist.karma.js
Comment (BrittanyIRL) [inline assets/src/edit-story/components/checklist/karma/checklist.karma.js]: great catch!

File: assets/src/edit-story/components/checklist/karma/checklist.karma.js
Comment (BrittanyIRL) [inline assets/src/edit-story/components/checklist/karma/checklist.karma.js]: 🤔  this is an incredibly long sleep, do we really need this?

Comment (samwhale) [reply assets/src/edit-story/components/checklist/karma/checklist.karma.js]: OOPS, nice catch. I was using that to test 😓

**Review by BrittanyIRL — APPROVED** [review]
🎉  

This is great! It looks like it'll be all good to go once #8204 is merged

File: assets/src/edit-story/components/checklist/karma/checklist.karma.js
Comment (samwhale) [reply assets/src/edit-story/components/checklist/karma/checklist.karma.js]: These should work when enabled once this is merged: #8212

**Comment by BrittanyIRL** [comment]
@googlebot I consent.

File: assets/src/edit-story/components/checklist/karma/checklist.karma.js
Comment (maxyinger) [inline assets/src/edit-story/components/checklist/karma/checklist.karma.js:219]: I feel like I've had issues with this approach in the past, but if it's working here that's awesome. curious what I was doing differently when I tried it last 🤷

File: assets/src/edit-story/components/checklist/karma/checklist.karma.js
Comment (embarks) [inline assets/src/edit-story/components/checklist/karma/checklist.karma.js:65]: thanks for the comments

File: assets/src/edit-story/components/checklist/karma/checklist.karma.js
Comment (embarks) [inline assets/src/edit-story/components/checklist/karma/checklist.karma.js:214]: we might want to introduce some count break here here so it doesn't hang in the `while`

**Review by embarks — APPROVED** [review]
One small thing about the while loop but otherwise the tests are great and easy to read.

File: assets/src/edit-story/components/checklist/karma/checklist.karma.js
Comment (BrittanyIRL) [reply assets/src/edit-story/components/checklist/karma/checklist.karma.js:214]: i'll do that in my PR that claudio is testing.

**Review by maxyinger — APPROVED** [review]
I'm good with everything here. Waiting to see results from the latest run in CI, but assuming it all passes, the tests themselves look good.

**Pull request merged**
